lib/Backend/SccLiveness.cpp

```diff
@@ -710,7 +754,11 @@ SCCLiveness::ExtendLifetime(Lifetime *lifetime, IR::Instr *instr)
         // The case of equality is valid on Arm64 where some branch instructions have sources.
         AssertMsg(lifetime->end >= instr->GetNumber(), "Lifetime end not set correctly");
 #else
-        AssertMsg(lifetime->end > instr->GetNumber(), "Lifetime end not set correctly");
+        AssertMsg(
+            (!instr->HasLazyBailOut() && lifetime->end > instr->GetNumber()) ||
+            (instr->HasLazyBailOut() && lifetime->end >= instr->GetNumber()),
```

Comment:
A comment explaining why its okay for the lifetime to end at the current instr if the instr has a lazy bailout would be nice.